gobblin-metrics/src/main/java/gobblin/metrics/event/sla/SlaEventSubmitter.java

```diff
@@ -0,0 +1,86 @@
+package gobblin.metrics.event.sla;
+
+import java.util.Map;
+import java.util.Properties;
+
+import lombok.Builder;
+import lombok.extern.slf4j.Slf4j;
+
+import org.apache.commons.lang.StringUtils;
+
+import com.google.common.base.Preconditions;
+import com.google.common.base.Predicate;
+import com.google.common.base.Strings;
+import com.google.common.collect.Maps;
+
+import gobblin.metrics.event.EventSubmitter;
+
+/**
+ * A wrapper around the {@link EventSubmitter} which can submit SLA Events.
+ */
+@Builder
+@Slf4j
+public class SlaEventSubmitter {
+
+  private EventSubmitter eventSubmitter;
+  private String eventName;
+  private String datasetUrn;
+  private String partition;
+  private String originTimestamp;
+  private String upstreamTimestamp;
+  private String recordCount;
+  private String previousPublishTimestamp;
+  private String dedupeStatus;
+  private String completenessPercentage;
+
+  public static SlaEventSubmitter of(String eventName, EventSubmitter eventSubmitter, Properties props) {
+    return SlaEventSubmitter.builder().eventName(eventName).eventSubmitter(eventSubmitter)
+        .datasetUrn(SlaEventKeys.DATASET_URN_KEY).partition(SlaEventKeys.PARTITION_KEY)
+        .originTimestamp(SlaEventKeys.ORIGIN_TS_KEY).upstreamTimestamp(SlaEventKeys.UPSTREAM_TS_KEY)
+        .completenessPercentage(SlaEventKeys.COMPLETENESS_PERCENTAGE_KEY).recordCount(SlaEventKeys.RECORD_COUNT_KEY)
+        .previousPublishTimestamp(SlaEventKeys.PREVIOUS_PUBLISH_TS_KEY).dedupeStatus(SlaEventKeys.DEDUPE_STATUS_KEY)
+        .build();
+  }
+
+  public void submit() {
```

Comment:
For additional metadata I was thinking of adding a member to the SlaEventSubmitter. You can set it using the builder. This field can be reused by the constructor that uses props to build the SlaEventSubmitter. We can have some helper methods to serialize any map into props.